dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/DotContentletTransformer.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    /**
+     * Main constructor provides access to set the required APIs
+     * @param contentlets input
+     * @param strategyResolver strategyResolver
+     * @param includeOptions
+     */
+    @VisibleForTesting
+    DotContentletTransformer(final List<Contentlet> contentlets,
+            final StrategyResolver strategyResolver,
+            final Set<TransformOptions> includeOptions) {
+        if(!isSet(contentlets)){
+           throw new DotRuntimeException("At least 1 contentlet must be set.");
+        }
+        this.contentlets = contentlets;
+        this.strategyResolver = strategyResolver;
+        this.includedOptions = includeOptions;
+    }
+
+    /**
+     * If desired we can do bulk transformation over a collection. So That's why we have a toMaps
+     * @return List of transformed Maps
+     */
+    public List<Map<String, Object>> toMaps() {
+        return contentlets.stream().map(this::copy).map(this::transform).collect(CollectionsUtils.toImmutableList());
+    }
+
+    /**
+     * This is the main method where individual transformation takes place.
+     * @param contentlet input contentlet
+     * @return Map holding the transformed properties
+     */
+    private Map<String, Object> transform(final Contentlet contentlet) {
+        final ContentType type = contentlet.getContentType();
+        if(null != type) {
+            Logger.debug(DotContentletTransformer.class,()->String.format(" BaseType: `%s` Type: `%s`", type.name(), type.baseType().name()));
+        }
+        final Map<String, Object> map = contentlet.getMap();
+        final List<AbstractTransformStrategy> strategies = strategyResolver.resolveStrategies(type);
+        for(final AbstractTransformStrategy strategy:strategies){
+           strategy.apply(contentlet, map, includedOptions);
+        }
+
+        return map;
+    }
+
+    /**
+     * Adds needed things that are not included by default from the api to the contentlet.
+     * If there is anything new to add, returns copy with the new attributes inside, otherwise returns the same instance.
+     * @return Contentlet returns a contentlet, if there is something to add will create a new instance based on the current one in the parameter and the new attributes
+     */
+    public List<Contentlet> hydrate() {
+        return contentlets.stream().map(this::copy).map(newContentlet -> {
+            transform(newContentlet);
+            return newContentlet;
+        }).collect(Collectors.toList());
+    }
+
+    /**
+     * To avoid caching issues we work on a copy
+     * @param contentlet input contentlet
+     * @return a copy contentlet
+     */
+    private Contentlet copy(final Contentlet contentlet) {
+       return TransformToolBox.copyContentlet(contentlet);
+    }
+
+    public static class Builder {
+
+         private List<Contentlet> contentlets = new ArrayList<>();
+
+         private Set<TransformOptions> optionsHolder = new HashSet<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Private field 'optionsHolder' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)